src/finn/custom_op/fpgadataflow/rtl/thresholding_rtl.py

```diff
@@ -0,0 +1,794 @@
+        istream_name = io_names["s_axis"][0][0]
+        ostream_name = io_names["m_axis"][0][0]
+        io_dict = {
+            "inputs": {istream_name: inp},
+            "outputs": {ostream_name: []},
+        }
+        self.rtlsim_multi_io(sim, io_dict)
+        output = io_dict["outputs"][ostream_name]
+
+        # Manage output data
+        odt = self.get_output_datatype()
+        target_bits = odt.bitwidth()
+        packed_bits = self.get_outstream_width()
+        out_npy_path = "{}/output.npy".format(code_gen_dir)
+        out_shape = self.get_folded_output_shape()
+
+        rtlsim_output_to_npy(output, out_npy_path, odt, out_shape, packed_bits, target_bits)
+
+        # load and reshape output
+        output = np.load(out_npy_path)
+        oshape = self.get_normal_output_shape()
+        output = np.asarray([output], dtype=np.float32).reshape(*oshape)
+        context[node.output[0]] = output
+        return
+
+    def rtlsim_multi_io(self, sim, io_dict):
+        "Run rtlsim for this node, supports multiple i/o streams."
+
+        rtlsim_so = self.get_nodeattr("rtlsim_so")
+        so_dir = os.path.dirname(os.path.realpath(rtlsim_so))
+        olcwd = os.getcwd()
+        os.chdir(so_dir)
+
+        # signal name prefix
+        # TODO if the interface names on this component get standardized,
+        # it won't need its own rtlsim_multi_io variant anymore and can just
+        # use the base class one
+        sname = "_"
+
+        trace_file = self.get_nodeattr("rtlsim_trace")
+        if trace_file == "default":
+            trace_file = self.onnx_node.name + ".vcd"
+        num_out_values = self.get_number_output_values()
+        reset_rtlsim(sim)
+        total_cycle_count = rtlsim_multi_io(
+            sim,
+            io_dict,
+            num_out_values,
+            trace_file=trace_file,
+            sname=sname,
+            liveness_threshold=pyverilate_get_liveness_threshold_cycles(),
+        )
+        self.set_nodeattr("cycles_rtlsim", total_cycle_count)
+        os.chdir(olcwd)
+
+    def code_generation_ipi(self):
+        """Constructs and returns the TCL commands for node instantiation as an RTL
+        block."""
+        rtl_file_list = [x.replace("template", self.onnx_node.name) for x in self.get_rtl_file_list()]
+        code_gen_dir = self.get_nodeattr("code_gen_dir_ipgen")
+        source_target = "./ip/verilog/rtl_ops/%s" % self.onnx_node.name
+        cmd = ["file mkdir %s" % source_target]
+
+        for rtl_file in rtl_file_list:
+            cmd.append(
+                "add_files -copy_to %s -norecurse %s"
+                % (source_target, os.path.join(code_gen_dir, rtl_file))
+            )
+
+        # Create an RTL block, not an IP core (-type ip)
+        cmd.append(
+            "create_bd_cell -type module -reference %s %s"
+            % (self.get_nodeattr("gen_top_module"), self.onnx_node.name)
+        )
+
+        return cmd
+
+    def get_verilog_top_module_intf_names(self):
```

Comment:
This looks identical to the function in the HLS variant and can likely be moved to `thresholding.py`.